shared/src/utils/graph.ts

```diff
@@ -372,6 +373,69 @@ export class RandomGraph {
       })
     })
 
+    // function ensureConnectedGraph(edges: EdgeList, nodes: NodeList, directed: boolean): void {
```

Comment:
The resulting graph ended up really messy and hard to read (The last node in a row of the grid connecting to the first node in the following row). Since it wasn't a priority stopped working on it for the moment.

I don't think I encountered an extractGiantComponent function